lib/Core/SQLiteBuildDB.cpp

```diff
@@ -600,6 +577,70 @@ class SQLiteBuildDB : public BuildDB {
   }
 
   /// @}
+
+  
+private:
+  /// Inserts a key if not present and always returns keyID
+  /// Sometimes key will be inserted for a lookup operation
+  /// but that is okay because it'll be added at somepoint anyway
```

Comment:
This second part of the comment isn't true anymore, I guess?

Reply:
Well, it is still technically true for this method. We no longer do it for 'lookupRuleResult', but it could conceivably happen for discovered dependencies.

Reply:
👍